.github/workflows/R-CMD-check.yaml

```diff
@@ -0,0 +1,37 @@
+# Run CI for R using https://eddelbuettel.github.io/r-ci/
+
+name: ci
+
+on:
+  push:
+  pull_request:
+
+env:
+  _R_CHECK_FORCE_SUGGESTS_: "false"
+
+jobs:
+  ci:
+    strategy:
+      matrix:
+        include:
+          - {os: macOS-latest}
+          - {os: ubuntu-latest}
+          - {os: windows-latest}
```

Comment:
As far as I know r-ci does not support Windows though.  It will happily do ubuntu-latest (these days leaning on r2u for everything CRAN) as well as macos-latest (using standard CRAN binaries) but I don't think I ever bothered with Windows.  If it works, great, but news to me :)

Reply:
this repository needed some love.  It was still set up with Travis-CI.

Reply:
so, the answer is, Windows doesn't work, but it fails gracefully.  will probably need to use a different image if we want to test Windows too.  I've merged the PR, and github Actions have now taken over for the deprecated Travis CI

Reply:
The replacement script put in almost correct -- see a few dozen repos of mine, or the [r-ci](https://eddelbuettel.github.io/r-ci) site -- but it this (simple. easy, reliable) framework simply does not concern itself with Windows.  And that was my one comment.

Reply:
(Didn't see your second comment while I wrote mine)

Correct, it will fail as there is no Windows support.  So my suggestion, again, is to remove that line from the matrix. 

The few times I think I may need Windows I just use a different yaml file for Windows.  I think across my many repos I do that _exactly once_.  For me, the use of CI to catch basic issues so I typically only run on Linux given that my code rarely has architecture-specific bits.  

Anyway, your repo, your call. I only replied as I was tagged and thought the comment would help. I'll remove myself from the conversation now.

Reply:
Thank you for noticing this, @eddelbuettel ! After I tested on my GH, I overlooked removing the tag. I'll explore the r-ci more in depth with the helpful docs you've provided.  Also,  `r-ci` handles the homebrew issues on macOS flawlessly, which was a bit of a vexing issue in the recent past, nicely done!!

Brian, I'll remove the windows tag so others (including our future selves) do not believe building blotter on Windows is passing continuous integration tests. On that note, I did send this to win-builder.r-project.org and it passed with 1 note concerning ["(possibly) invalid URLs"](https://win-builder.r-project.org/L03GkXRAS13g/00check.log), which I will fix. 

Thanks for merging this PR @braverock, as I went a little further than the scope of pending issues! These fixes enhance package stability and, hopefully, will keep the `blotter` pkg building clean for another few years.

Reply:
@JustinMShea I am actually mostly unsure how macOS here.  It is more an afterthought.  It does take the img file from CRAN (so not brew / homebrew) and installs the regular way via `install.packages` (hence also CRAN package binaries, not brew / homebrew).  At least I think so :)  

I actually enhanced `r-ci` yesterday and today. I realized the `./run.sh bootstrap` can be made redundant by starting from an action definition that includes it, so I made one.  And I recently ran into a (rare, for me) use on macOS that also needed `gfortran` (so that BLAS/LAPACK/... work for `RcppArmadillo`) so I added that ... this morning.   I will try to update the webpage and blog about it tomorrow.   Glad you find it useful.